allennlp/semparse/worlds/atis_world.py

```diff
@@ -31,74 +59,182 @@ def __init__(self,
                  utterances: List[str],
                  tokenizer: Tokenizer = None,
                  database_directory: str = None) -> None:
+
+        self.all_tables = ALL_TABLES
+        self.tables_with_strings = TABLES_WITH_STRINGS
         if database_directory:
-            self.sql_table_context = SqlTableContext(ALL_TABLES,
-                                                     TABLES_WITH_STRINGS,
-                                                     database_directory)
+            self.database_directory = database_directory
+            self.connection = sqlite3.connect(self.database_directory)
+            self.cursor = self.connection.cursor()
+
+        self.sql_table_context = SqlTableContext(ALL_TABLES,
+                                                 TABLES_WITH_STRINGS,
+                                                 database_directory,
+                                                 utterances=utterances) if database_directory else None
```

Comment:
It doesn't look like you use `sql_table_context.utterances` here, either; I'm not sure why you added it, because you already save it to `AtisWorld`.